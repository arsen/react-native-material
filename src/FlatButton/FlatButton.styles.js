import { StyleSheet, Dimensions, Platform } from 'react-native';
import deepAssign from 'deep-assign';
import Color from 'color';

export default {

  container: (theme, props) => {
    let styles = {
      backgroundColor: theme.backgroundColor,
    };
    return deepAssign(styles, props.style);
  },

  innerStyle: () => {
    return {
      paddingLeft: 16,
      paddingRight: 16,
      paddingTop: 8,
      paddingBottom: 8,
      justifyContent: 'center',
      alignItems: 'center',
      flexDirection: 'row'
    };
  },

  labelStyle: (theme, props) => {
    let color = props.disabled ? theme.disabledLabelColor :
      props.primary ? theme.primaryLabelColor :
        props.secondary ? theme.secondaryLabelColor :
          theme.labelColor;

    let styles = {
      fontSize: 14,
      color: color,
    };
    return deepAssign(styles, props.labelStyle);
  },
};