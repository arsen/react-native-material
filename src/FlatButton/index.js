import component from './FlatButton';
export default component;