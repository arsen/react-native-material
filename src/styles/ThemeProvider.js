import {Component, PropTypes} from 'react';
import deepAssign from 'deep-assign';

import defaultTheme from './themes/default';

class ThemeProvider extends Component {

  static propTypes = {
    children: PropTypes.element,
    theme: PropTypes.object,
  };

  static defaultProps = {
    theme: {}
  };

  static childContextTypes = {
    theme: PropTypes.object.isRequired,
  };

  getChildContext() {
    return {
      theme: deepAssign(defaultTheme, this.props.theme)
    };
  }

  render() {
    return this.props.children;
  }
}

export default ThemeProvider;