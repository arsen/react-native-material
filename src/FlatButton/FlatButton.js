import React, { Component, PropTypes } from 'react';
import {
  View,
  Text,
} from 'react-native';

import Button from '../internal/Button';

import styles from './FlatButton.styles';

/**
 * FlatButton Component.
 */
export default class FlatButton extends Component {
  constructor(props) {
    super(props);
  }

  static contextTypes = {
    theme: PropTypes.object.isRequired,
  }

  static propTypes = {
    /**
     * Label for the button.
     */
    label: PropTypes.string.isRequired,

    /**
     * Callback function when button is pressed.
     */
    onPress: PropTypes.func,

    /**
     * Callback function when button is long pressed.
     */
    onLongPress: PropTypes.func,

    /**
     * Override button container styles.
     */
    style: PropTypes.object,

    /**
     * If true, colors button according to primary color from the Theme
     */
    primary: PropTypes.bool,

    /**
     * If true, colors button according to secondary color from the Theme
     */
    secondary: PropTypes.bool,

    /**
     * 	Disables the button if set to true.
     */
    disabled: PropTypes.bool,
  }

  static defaultProps = {
    onPress: () => { },
    onLongPress: null,
    style: {},
    labelStyle: {},
    primary: false,
    secondary: false,
    disabled: false,
  }

  render() {
    const theme = this.context.theme.FlatButton;
    const props = this.props;
    return (
      <Button 
        label={props.label}
        style={styles.container(theme, props)}
        innerStyle={styles.innerStyle()}
        labelStyle={styles.labelStyle(theme, props)}
        disabled={props.disabled}
        onPress={props.onPress}
        onLongPress={props.onLongPress}
      />
    );
  }
}