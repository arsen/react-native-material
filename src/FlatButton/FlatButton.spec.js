import 'react-native';
import React from 'react';
import ThemeProvider from '../styles/ThemeProvider';
import FlatButton from './FlatButton';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Flatbutton renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider>
      <FlatButton label="Flat Button" />
    </ThemeProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
