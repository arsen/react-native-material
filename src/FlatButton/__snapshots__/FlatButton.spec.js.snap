exports[`test Flatbutton renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0)",
    }
  }>
  <View
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 2,
          "overflow": "hidden",
        },
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
          },
        ],
      ]
    }>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 14,
          }
        }>
        Flat Button
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.04349999999999998)",
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      } />
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.08699999999999997)",
          "borderRadius": 2000,
          "bottom": 0,
          "height": 0,
          "left": 0,
          "marginLeft": 0,
          "marginTop": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      } />
  </View>
</View>
`;
